### DISCUSSION TITLE
Create workitem trough RestFul

### DISCUSSION THREAD

**Discussion by jmreynam** [dcm4che/dcm4chee-arc-light#3853] [Q&A] [answered]
Hello, I'm triying to create a workitem via HTTP Requests, I'm using Postman with GET request to bring the whole worklist and I don't have any problems, but when I use POST request Postman show me Error 406. The link for the requests is the following http://ip:8080/dcm4chee-arc/aets/DCM4CHEE/rs/workitems/. Is there any way for create a Workitem trough this method?

Thanks for your help.

**Comment by vrindanayak** [comment]
Have you passed the required payload to [create new workitem](https://petstore.swagger.io/index.html?url=https://dcm4che.github.io/dcm4chee-arc-light/swagger/openapi.json#/UPS-RS/CreateWorkitem)?

Alternatively, you can use [upsrs tool](https://github.com/dcm4che/dcm4che20/tree/master/tool/upsrs) which simulates invocation of [UPS-RS services](https://petstore.swagger.io/index.html?url=https://dcm4che.github.io/dcm4chee-arc-light/swagger/openapi.json#/UPS-RS)

**Comment by jmreynam** [comment]
Hello, thanks for your answer. I don't have idea about how pass the required paylod to create new workitem, there is some documentation to follow the steps?
Do I have to read the documentation of dicom.nema?

Thanks.

**Comment by vrindanayak** [comment]
You may use archive UI to create new workitem and look at the request payload json in browser inspect.

**Comment by jmreynam** [comment]
Thanks @vrindanayak, I can solve this problem. Finally, How delete the workitem?

**Comment by vrindanayak** [comment, marked as answer]
Once the workitem state changes to COMPLETED or CANCELED and if the workitems do not have any deletion locks set on them, archive's default configurations ([Delete UPS Polling Interval](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-assembly/src/main/resources/ldap/default-config.ldif#L119) / [Delete UPS Completed Delay](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-assembly/src/main/resources/ldap/default-config.ldif#L91) / [Delete UPS Canceled Delay](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-assembly/src/main/resources/ldap/default-config.ldif#L38)) ensure to delete these workitems.

See [DICOM part4 - Change UPS State SCP behaviour](https://dicom.nema.org/medical/dicom/current/output/html/part04.html#sect_CC.2.1.3)
> After the state of the UPS instance has been changed to COMPLETED or CANCELED, the SCP shall not delete the instance until all deletion locks have been removed.

**Comment by jmreynam** [comment]
Thanks for your help @vrindanayak, I am very grateful. I can solve this, I spent days trying to solve it.
